broker/api_test.go

```diff
@@ -0,0 +1,494 @@
+package broker_test
+
+import (
+	"code.cloudfoundry.org/lager"
+	composeapi "github.com/compose/gocomposeapi"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/alphagov/paas-compose-broker/broker"
+	"github.com/alphagov/paas-compose-broker/catalog"
+	"github.com/alphagov/paas-compose-broker/compose/fakes"
+	"github.com/alphagov/paas-compose-broker/config"
+	"github.com/pivotal-cf/brokerapi"
+	uuid "github.com/satori/go.uuid"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+)
+
+func NewRequest(method, path string, body io.Reader, username, password string, params ...UriParam) *http.Request {
+	brokerUrl := fmt.Sprintf("http://%s", "127.0.0.1:8080"+path)
+	req := httptest.NewRequest(method, brokerUrl, body)
+	if username != "" {
+		req.SetBasicAuth(username, password)
+	}
+	q := req.URL.Query()
+	for _, p := range params {
+		q.Add(p.Key, p.Value)
+	}
+	req.URL.RawQuery = q.Encode()
+	return req
+}
+
+type UriParam struct {
+	Key   string
+	Value string
+}
+
+func ReadResponseBody(responseBody *bytes.Buffer) []byte {
+	body, err := ioutil.ReadAll(responseBody)
+	Expect(err).ToNot(HaveOccurred())
+	return body
+}
+
+func DoRequest(server http.Handler, req *http.Request) *httptest.ResponseRecorder {
+	w := httptest.NewRecorder()
+	server.ServeHTTP(w, req)
+	return w
+}
+
+var Yes = true
```

Comment:
Heh, these were bits of debuging shrapnel left around and shouldn't be there. Will remove.